### PR TITLE
Tom/DLS-8749 Updated calculator show the rebase calculation

### PR DESCRIPTION
**Pull request by TomRafferty**
This is the functional side of this ticket and adds:
- updates to the calculator addressing an edge case and implementing the visibility of rebase calculations
- tests for the CalculatedGlarBreakdown model as well as UI tests to check visibility of the rebase calculations in working out drop down

Given I have enough time - I'm going to be working on a refactor for this as I'm not happy with the quality of the tests - however to reduce the possibility of this work taking to long, this PR is strictly for functionalities sake.

### CODE REVIEW

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        breakdown.totalReliefs shouldBe AmountInPence(600000)
+      }
+      "Return an accurate initialGainOrLoss value" in {
+        breakdown.initialGainOrLoss shouldBe AmountInPence(5900000)
+      }
+      "Return an accurate acquisitionOrRebased value" in {
+        breakdown.acquisitionOrRebased shouldBe AmountInPence(30000000)
+      }
+      "Return an accurate gainOrLossAfterReliefs value" in {
+        breakdown.gainOrLossAfterReliefs shouldBe AmountInPence(5300000)
```
Comment (LAJW) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala:75]: ⛏️ Missing large number separators.

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        acquisitionCosts = AmountInPence(15_000_00),
+        disposalPrice = AmountInPence(400_000_00),
+        disposalFees = AmountInPence(16_000_00),
+        privateResidentReliefs = AmountInPence(5_000_00),
+        lettingRelief = AmountInPence(1_000_00),
+        improvementCosts = AmountInPence(10_000_00),
+        shouldUseRebase = true,
+        rebasedAcquisitionPrice = None
+      )
+      "Throw an accurate exception" in {
```
Comment (LAJW) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala:184]: ⛏️ I wouldn't call it "accurate exception", just "throw `Exception`". I'd state somewhere that this is an edge case and wouldn't normally happen.

**Review by LAJW — APPROVED** [review]

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala (hunk 1 above)
Comment (TomRafferty) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala]: resolved

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala [hunk 3]
```diff
@@ -57,26 +57,26 @@ class CalculatedGlarBreakdownSpec extends AnyWordSpec with Matchers {
...
+        breakdown.acquisitionOrRebased shouldBe AmountInPence(300_000_00)
       }
       "Return an accurate gainOrLossAfterReliefs value" in {
-        breakdown.gainOrLossAfterReliefs shouldBe AmountInPence(5300000)
+        breakdown.gainOrLossAfterReliefs shouldBe AmountInPence(53_000_00)
       }
 
     }
-    "Given a loosing disposal" should {
+    "Given a loosing disposal"                               should {
```
Comment (LAJW) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala:79]: ⛏️ Why did you add these spaces here?

**Review by LAJW — APPROVED** [review]

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala (hunk 3 above)
Comment (TomRafferty) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala:79]: I believe these are from bad paste jobs - will resolve now (thought I'd caught them all)

Comment (TomRafferty) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala]: should be resolved now 👍

File: test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala [hunk 4]
```diff
@@ -107,7 +107,7 @@ class CalculatedGlarBreakdownSpec extends AnyWordSpec with Matchers {
         breakdown.gainOrLossAfterReliefs shouldBe AmountInPence(-135_000_00)
       }
     }
-    "using a rebase value" should {
+    "using a rebase value"                                   should {
```
Comment (LAJW) [inline test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala:110]: There's still one left.

Comment (TomRafferty) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala:110]: 🤦

Comment (TomRafferty) [reply test/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/returns/CalculatedGlarBreakdownSpec.scala]: should be good now - apologies
